### ISSUE TITLE
Proxy issues - Bad Avocode

### ISSUE THREAD

**Issue by marklee227** [avocode/avocode#197]
I downloaded the app and launched it. Once I click the "Explore Avocode" button, I get this message. "Your Avocode has gone bad"

**Comment by petrbrzek** [comment]
Not very pleasant exploring. Does this happen every time you open Avocode? What is your OS system?

**Comment by marklee227** [comment]
Yes, it happens every time. Does this have to do with network proxy?

OS:
10.9.5

**Comment by marklee227** [comment]
My coworker installed Avocode and could not get it to work. I believe it’s our network proxy at work.

**Comment by petrbrzek** [comment]
Can you please try this command in your terminal?

- Change `http://127.0.0.1:9999` to your proxy address.
- Check your application path - this command supposes that Avocode app is in Applications folder

```
http_proxy="http://127.0.0.1:9999" https_proxy="http://127.0.0.1:9999" /Applications/Avocode.app/Contents/MacOS/Atom --proxy-server="http://127.0.0.1:9999"
```

**Comment by krumthi** [comment]
So I am working with Marklee227 on the issue and I am still getting the error your Avocode has gone bad. But I do see the following in the terminal window 
![screen shot 2015-04-20 at 1 54 10 pm](https://cloud.githubusercontent.com/assets/12039168/7238867/279ff2b2-e76b-11e4-8b77-2a9c0f6ba046.png)

we are under an authenticated firewall and using a pac file. Please help if possible.

**Comment by rychlis** [comment]
As petr mentioned, you need to replace `http://127.0.0.1:9999` to your proxy address which should be inside the PAC file.

**Comment by krumthi** [comment]
![screen shot 2015-04-24 at 9 52 18 am](https://cloud.githubusercontent.com/assets/12039168/7322344/16428408-ea6d-11e4-8733-77f9b78fa75d.png)
![screen shot 2015-04-24 at 10 32 30 am](https://cloud.githubusercontent.com/assets/12039168/7322365/3e681f2e-ea6d-11e4-9107-4c25416ae5d8.png)

So  I still get the same error message when I do this  I have tried two different versions of our proxy, I have a direct line with our internet team so please let me know what else I need to do 

Thank you

**Comment by krumthi** [comment]
Anyone working on this at all? Anyone?

**Comment by petrbrzek** [comment]
We're upgrading Electron (internal framework). That should fix it. I'll let you know when we release it.

**Comment by krumthi** [comment]
Do we have a time Frame on when this will be completed?

**Comment by petrbrzek** [comment]
We should release the new version next week. That's the most accurate guess I can give now.

**Comment by marklee227** [comment]
Upon William’s request, I installed the latest Avocode and it appears to be working now. The new version is 2.1.0.

**Comment by krumthi** [comment]
This is Not closed at all. the issue still remains Marklee was wrong on doing this. Please reopen this case.

**Comment by marklee227** [comment]
Were you able to create and account and log in? Did you upload your own file to test it putt?

I have the most updated version and am still not able to access.


From: <Lee>, Mark Lee <mark.lee@transamerica.com<mailto:mark.lee@transamerica.com>>
Date: Tuesday, May 19, 2015 at 12:22 PM
To: avocode/avocode <reply@reply.github.com<mailto:reply@reply.github.com>>, avocode/avocode <avocode@noreply.github.com<mailto:avocode@noreply.github.com>>
Cc: "Wu, William" <William.Wu@transamerica.com<mailto:William.Wu@transamerica.com>>, Erik Krumholz <erik.krumholz@transamerica.com<mailto:erik.krumholz@transamerica.com>>, Ben Suarez <Ben.Suarez@transamerica.com<mailto:Ben.Suarez@transamerica.com>>
Subject: Re: [avocode] Bad Avocode (#197)

Upon William’s request, I installed the latest Avocode and it appears to be working now. The new version is 2.1.0.

**Comment by marklee227** [comment]
My mistake. The program opened up this time. That’s all I did. I didn’t do any testing on it yet. Thanks for reopening it.



From: <Suarez>, Ben Suarez <ben.suarez@transamerica.com<mailto:ben.suarez@transamerica.com>>
Date: Wednesday, May 20, 2015 at 9:12 AM
To: Mark Lee <mark.lee@transamerica.com<mailto:mark.lee@transamerica.com>>, avocode/avocode <reply@reply.github.com<mailto:reply@reply.github.com>>, avocode/avocode <avocode@noreply.github.com<mailto:avocode@noreply.github.com>>
Cc: "Wu, William" <William.Wu@transamerica.com<mailto:William.Wu@transamerica.com>>, Erik Krumholz <erik.krumholz@transamerica.com<mailto:erik.krumholz@transamerica.com>>
Subject: Re: [avocode] Bad Avocode (#197)

Were you able to create and account and log in? Did you upload your own file to test it putt?

I have the most updated version and am still not able to access.


From: <Lee>, Mark Lee <mark.lee@transamerica.com<mailto:mark.lee@transamerica.com>>
Date: Tuesday, May 19, 2015 at 12:22 PM
To: avocode/avocode <reply@reply.github.com<mailto:reply@reply.github.com>>, avocode/avocode <avocode@noreply.github.com<mailto:avocode@noreply.github.com>>
Cc: "Wu, William" <William.Wu@transamerica.com<mailto:William.Wu@transamerica.com>>, Erik Krumholz <erik.krumholz@transamerica.com<mailto:erik.krumholz@transamerica.com>>, Ben Suarez <Ben.Suarez@transamerica.com<mailto:Ben.Suarez@transamerica.com>>
Subject: Re: [avocode] Bad Avocode (#197)

Upon William’s request, I installed the latest Avocode and it appears to be working now. The new version is 2.1.0.

**Comment by marklee227** [comment]
I’m pretty sure what you saw was the example file that give you.

Erik and I discussed and it is indeed not working yet.

From: <Lee>, Mark Lee <mark.lee@transamerica.com<mailto:mark.lee@transamerica.com>>
Date: Wednesday, May 20, 2015 at 12:49 PM
To: Ben Suarez <Ben.Suarez@transamerica.com<mailto:Ben.Suarez@transamerica.com>>, avocode/avocode <reply@reply.github.com<mailto:reply@reply.github.com>>, avocode/avocode <avocode@noreply.github.com<mailto:avocode@noreply.github.com>>
Cc: "Wu, William" <William.Wu@transamerica.com<mailto:William.Wu@transamerica.com>>, Erik Krumholz <erik.krumholz@transamerica.com<mailto:erik.krumholz@transamerica.com>>
Subject: Re: [avocode] Bad Avocode (#197)

My mistake. The program opened up this time. That’s all I did. I didn’t do any testing on it yet. Thanks for reopening it.



From: <Suarez>, Ben Suarez <ben.suarez@transamerica.com<mailto:ben.suarez@transamerica.com>>
Date: Wednesday, May 20, 2015 at 9:12 AM
To: Mark Lee <mark.lee@transamerica.com<mailto:mark.lee@transamerica.com>>, avocode/avocode <reply@reply.github.com<mailto:reply@reply.github.com>>, avocode/avocode <avocode@noreply.github.com<mailto:avocode@noreply.github.com>>
Cc: "Wu, William" <William.Wu@transamerica.com<mailto:William.Wu@transamerica.com>>, Erik Krumholz <erik.krumholz@transamerica.com<mailto:erik.krumholz@transamerica.com>>
Subject: Re: [avocode] Bad Avocode (#197)

Were you able to create and account and log in? Did you upload your own file to test it putt?

I have the most updated version and am still not able to access.


From: <Lee>, Mark Lee <mark.lee@transamerica.com<mailto:mark.lee@transamerica.com>>
Date: Tuesday, May 19, 2015 at 12:22 PM
To: avocode/avocode <reply@reply.github.com<mailto:reply@reply.github.com>>, avocode/avocode <avocode@noreply.github.com<mailto:avocode@noreply.github.com>>
Cc: "Wu, William" <William.Wu@transamerica.com<mailto:William.Wu@transamerica.com>>, Erik Krumholz <erik.krumholz@transamerica.com<mailto:erik.krumholz@transamerica.com>>, Ben Suarez <Ben.Suarez@transamerica.com<mailto:Ben.Suarez@transamerica.com>>
Subject: Re: [avocode] Bad Avocode (#197)

Upon William’s request, I installed the latest Avocode and it appears to be working now. The new version is 2.1.0.

**Comment by krumthi** [comment]
Still waiting on that update to come out I hope it comes out sooner then later

**Comment by petrbrzek** [comment]
I'm sorry for the delay. The good news is that we solved those proxy issues in our dev build. I don't want to promise the release date but it's really close.

**Comment by krumthi** [comment]
I am guessing that the delay is still going on since I haven't heard anything about an update recently.

**Comment by rychlis** [comment]
We've finally released Avocode 2.2.0. Sorry about the delay.

To use proxy server, see [this article](https://avocode.zendesk.com/hc/en-us/articles/203402352)

**Comment by krumthi** [comment]
![screen shot 2015-06-16 at 3 29 22 am](https://cloud.githubusercontent.com/assets/12039168/8178885/e7d68e04-13d7-11e5-8eaf-3ce1b182025a.png)


We have gotten further, but we still can not sync anything, I just tried to send up this test file and it is giving this error message
"Something went wrong please try again". I did both of the commands that were inside your article. 

So again this is still not closed. 
 Before you close an issue you should always check with the people that are working with it.

**Comment by rychlis** [comment]
you can simply re-open the issue if you don't think it's closed...

Are all other features working? If there is now only error when uploading, I will create a new issue for that. You can upload PSDs using the [web manager](https://manager.avocode.com) until it's fixed

**Comment by krumthi** [comment]
I wouldn't know what other features that we are looking for.  All I know is that the sync feature doesn't update and I get very Vague errors

**Comment by rychlis** [comment]
Hello, we've just released Avocode 2.2.1 which should fix this issue. Let me know if it's ok now

**Comment by marklee227** [comment]
Thanks for the update. I have to test it out.

**Comment by krumthi** [comment]
I have been able to test out the product 2.2.1 which did work and solved the issues, but I do request that you need to find a way to make this permanent for 10.10 and 10.11 since the   /etc/launchd.conf long exists in these environments but for the time being you can close this out

**Comment by rychlis** [comment]
There doesn't seem to be a [easy solution](http://stackoverflow.com/a/25912736/823726) for setting permanent variables in Yosemite, we should probably add persistent proxy settings somewhere in the application settings. I've created a uservoice idea for this: [Proxy settings in application](https://avocode.uservoice.com/forums/277080-feature-voting/suggestions/8615353-proxy-settings-in-application)

I am closing the issue, thanks for testing.

**Comment by marklee227** [comment]
Thank you
